ext/nnstreamer/tensor_decoder/tensordec-imagesegment.c

```diff
@@ -0,0 +1,287 @@
+/**
+ * GStreamer / NNStreamer tensor_decoder subplugin, "image segment"
+ * Copyright (C) 2019 Jihoon Lee <ulla4571@gmail.com>
+ * Copyright (C) 2019 niklasjang <niklasjang@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	tensordec-directvideo.c
+ * @date	19 Oct 2019
+ * @brief	NNStreamer tensor-decoder subplugin, "image segment",
+ *              which detects objects and paints their regions.
+ *
+ * @see		https://github.com/nnsuite/nnstreamer
+ * @author  Jihoon Lee <ulla4571@gmail.com>
+ *          niklasjang <niklasjang@gmail.com>
+ * @bug		No known bugs except for NYI items
+ *
+ * option1: Decoder mode of image segmentation
+ *          Available : tflite-deeplab
+ */
+
```

Comment:
It would be best if you describe an example pipeline that tests this decoder here as a comment along with a tensorfilter and model that gives the tensor stream that can be decoded by this decoder.

Reply:
Can we add example pipeline with specific file like this?
```
gst-launch-1.0 -v \
      filesrc location=cat.png ! decodebin ! videoconvert ! videoscale ! imagefreeze !\
      video/x-raw,format=RGB,width=257,height=257,framerate=10/1 ! tee name=t \
      t. ! tensor_converter !\
      tensor_transform mode=arithmetic option=typecast:float32,add:0.0,div:255.0 !\
      tensor_filter framework=tensorflow-lite model=deeplabv3_257_mv_gpu.tflite !\
      tensor_decoder mode=image_segment option1=tflite-deeplab !\
      compositor name=mix sink_0::zorder=2 sink_1::zorder=1 !\
      videoconvert ! autovideosink \
      t. ! queue ! mix.
```

Reply:
Just add brief pipeline with description.

Reply:
If you could actually run this pipeline, that is fine. Thanks!

Reply:
My pleasure :) gst string pipeline is updated.